libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/LargeCausalIndividualView/LargeCausalIndividualChart.tsx

```diff
@@ -153,8 +153,36 @@ export class LargeCausalIndividualChart extends React.PureComponent<
   };
 
   private updateBubblePlotData = async (
-    chartProps: IGenericChartProps
+    chartProps: IGenericChartProps,
+    hasRevertToBubbleChartUpdated: boolean
   ): Promise<void> => {
+    if (hasRevertToBubbleChartUpdated) {
+      this.setState(
+        {
+          chartProps,
+          clusterData: getInitialClusterState(),
+          isBubbleChartDataLoading: true,
+          isBubbleChartRendered: true,
+          isLocalCausalDataLoading: false,
+          isRevertButtonClicked: false,
+          localCausalData: undefined,
+          localCausalErrorMessage: undefined,
+          selectedPointsIndexes: []
+        },
+        () => {
+          this.setState({
+            chartProps,
+            isBubbleChartDataLoading: false,
+            isBubbleChartRendered: true,
```

Comment:
is `isBubbleChartRendered and isRevertButtonClicked` needed (line 174; 176; 177)? They are not changed right?

Reply:
It is like resetting the state. So whenever bubble chart is rendered, these has to be reset. isBubbleChartRendered could be false in case of scatter plot so it has to be reset to true once bubble plot is rendered. isRevertButtonClicked would be set to true when 'View bubble plot' button is clicked. THis has to be reset too.